src/components/NumberInput/NumberInput.jsx

```diff
@@ -1,57 +1,69 @@
 import { isNumber, toNumber } from 'lodash';
 import React from 'react';
 import PropTypes from 'prop-types';
+import cx from 'classnames';
 import style from './style.scss';
 import {
   getDataAttrs,
   getDataProps,
 } from '../../dataUtils';
 
 class NumberInput extends React.Component {
-
-  constructor() {
-    super();
-    this.onPressIncrement = this.onPressIncrement.bind(this);
-    this.onPressDecrement = this.onPressDecrement.bind(this);
-    this.onChange = this.onChange.bind(this);
-  }
-
-  onPressIncrement() {
-    if (!isNumber(this.props.maxValue) ||
-        this.props.maxValue >= this.props.value + this.props.step) {
-      this.props.onPressIncrement({ diff: this.props.step });
+  onPressIncrement = () => {
+    const { maxValue, value, step, onBlur, onChange } = this.props;
+    if (maxValue >= value + step) {
+      onChange(value + step);
+      onBlur && onBlur();
     }
   }
 
-  onPressDecrement() {
-    if (!isNumber(this.props.minValue) ||
-        this.props.minValue <= this.props.value - this.props.step) {
-      this.props.onPressDecrement({ diff: -1 * this.props.step });
+  onPressDecrement = () => {
+    const { minValue, value, step, onBlur, onChange } = this.props;
+    if (minValue <= value - step) {
+      onChange(value - step);
+      onBlur && onBlur();
     }
   }
 
-  onChange(event) {
-    const value = toNumber(event.nativeEvent.target.value);
-    if (
-      (!this.props.maxValue || value <= this.props.maxValue) &&
-      (!this.props.minValue || value >= this.props.minValue)
-    ) {
-      this.props.onChange({ value });
-    }
+  onChange = (event) => {
+    const { maxValue, minValue, onChange } = this.props;
+    const eventValue = event.nativeEvent.target.value;
+
+    // if the value is cleared, do not convert empty string to 0
+    const value = eventValue === "" ? null : toNumber(eventValue);
```

Comment:
we should probably add a prop called allowBlank that if the value is null, we skip these checks.